packages/ember-data/lib/system/relationships/state/relationship.js

```diff
@@ -216,5 +219,9 @@ Relationship.prototype = {
 
   setHasData: function(value) {
     this.hasData = value;
+  },
+
+  setHasLoaded: function(value) {
+    this.hasLoaded = value;
```

Comment:
`hasData` is "we consider the state of this relationship known"
`hasLoaded` is "the adapter has given us the content of this relationship"

They look very similar but have different purposes.

```
let post = store.createRecord('post', { id: 'dummy' });

// all relationships for post are now known, hasData = true,
// since the record was created locally. but hasLoaded = false

store.pushPayload('post', { id: 'dummy', links: { author: 'author/1' }});

// now let's get the author
post.get('author');

// since we do have a link, and hasLoaded is false,
// we know to fetch it even if hasData is true
```

Another example is when a relationship is known (hasData = true) and a link is updated (updateLink sets hasLoaded = false). This way ED know to fetch the link again, even though we have a known state.